### ISSUE TITLE
QuickLookup v1.0.2 - Display multiple fields in "I2 Which Field" not functioning properly.

### ISSUE THREAD

**Issue by bottomlinebrokers** [alexed1/LightningFlowComponents#1137]
I installed QuickLookup v1.0.2 into my Summer '22 sandbox development org.  I am attempting to use the features described here ( 
 https://unofficialsf.com/new-option-for-the-lookup-component/   ) to display multiple fields on a record in the Lookup dropdown while searching on the Name field.  
 

Per the page, I have to provide “I2_Display Which Field?”  a formula field with the multiple fields to display in the dropdown
and “I8_Search this Field instead of Display Field” the field to search on.

I've embedded the component in my screen flow (running under API 55).  The I8 portion works; I provide a fieldname (custom or standard) and Quick Lookup will search on it.  Providing I2 a single field name, or a formula (or text) field containing a SINGLE fieldname works as well.  It will search on the I2 entry, and display the I8 entry.

### **Steps to Reproduce**
Attempt to expand the formula for I2 to display multiple fields.  It simply displays nothing:  

![Error](https://user-images.githubusercontent.com/43708364/173702353-a225a1d4-a9ee-498a-a5d5-580b6ba54518.png)

I am attempting to search on the Opportunity object.  My formula field for I2 was:

"Name + Property_City__c + Listing_Number__c + StageName"
(Note: Name and Property_City are text fields.  StageName is a picklist.  Listing_Number__c is an autonumber field, but that should be text)

I also attempted these derivations in the formula:

"Name, Property_City__c"
"Name  Property_City__c"
"Name | Property_City__c"
"Name" + "Property_City__c"

I also tried these derivations in a Text field rather than a formula field

Name, Property_City__c
Name  Property_City__c
Name | Property_City__c
Name + Property_City__c

Please assist.  In this flow, our users will have to choose among several Opportunities that are named the same, but are in different cities or have a different Listing number.  

Thank you!

Mark

**Comment by ericrsmith35** [comment]
I'm assuming that you have created a formula field on the Opportunity object and are putting that field API name into I2.

That said, the formula should be "Name & Property_City__c & Listing_Number__c & TEXT(StageName)"

**Comment by bottomlinebrokers** [comment]
No, I created the formula field inside the flow and attempted to use that.  I didn't think I would need to store this as data on a record, as this type of display is only used in this flow.

Is this incorrect?
________________________________

**Comment by ericrsmith35** [comment]
The component is designed to work with an actual field on the record.

**Comment by bottomlinebrokers** [comment]
Well then, that will hopefully solve my problem...I will try now!
________________________________

**Comment by bottomlinebrokers** [comment]
Eric,

Using a formula field on the Opportunity object was also unsuccessful.

I attempted 3 formulae:

1 - "Name & Property_City__c & Listing_Number__c & TEXT(StageName)", see Error 1.png
2 - "Name " & "Property_City__c " & "Listing_Number__c " & "TEXT(StageName)", see Error2.png
3 - "Name," & "Property_City__c, " & "Listing_Number__c, " & "TEXT(StageName)", see Error3.png

![Error 1](https://user-images.githubusercontent.com/43708364/173883825-5cae17a2-6ea5-4947-b8fa-80a20e170d3f.png)
![Error 2](https://user-images.githubusercontent.com/43708364/173883828-80d69731-2cc8-43f8-a561-5e791884b3a7.png)
![Error 3](https://user-images.githubusercontent.com/43708364/173883832-7b456995-10cf-4c8b-a581-cd883130ab2d.png)

Next steps?

**Comment by ericrsmith35** [comment]
When you are creating your formula field on the Opportunity object, do not manually type in the field names, operators or functions.  Always use the Insert Field , Insert Operator and Insert Selected Function buttons so the syntax is correct.

**Comment by bottomlinebrokers** [comment]
The formula as your specified originally did not work.


Here is what did work, exact copy/paste:


Name & ", " & Property_City__c & ", " & Listing_Number__c & ", " & TEXT(StageName)


I chose ',' characters as delimiters, because they were most natural on the eye.


FYI, I upgraded to v1.0.3

Mark
________________________________

**Comment by alexed1** [comment]
What would be a good resolution for this? Is this something we should post about? Or just close it as a case of 'formulas can be tricky to get right'?

**Comment by bottomlinebrokers** [comment]
Alex,

No...it's not just "formulas can be tricky".  The formula format on the original page (https://unofficialsf.com/new-option-for-the-lookup-component/) does not match the formula format that I got to work.  So I suspect that the formula example on the page is incorrect.  I would suggest somebody edit the original page with 2 or 3 examples of formulae that actually work.  You can try mine above...they are in production now.

Thank you for following up!

Mark
